### PR TITLE
Conor/pml parsing

### PR DESCRIPTION
**Pull request by c-brenn**
minimal parser that can accept/reject pml. 

It correctly parses all of [jnoll's sample pml](https://github.com/jnoll/peos/tree/master/compiler/models) which is in `test/fixtures`. Don't know of a way to make github ignore that directory in the PR diff 👴

### CODE REVIEW

File: panacea/.gitignore [hunk 1]
```diff
@@ -22,3 +22,7 @@ erl_crash.dump
 # secrets file as long as you replace its contents by environment
 # variables.
 /config/prod.secret.exs
+
+# the generated parser and lexer modules
+/src/pml_lexer.erl
+/src/pml_parser.erl
```
Comment (houli) [inline panacea/.gitignore:28]: Should these also be added to the dockerignore?

**Review by houli — COMMENTED** [review]
How does the generation of the parser and lexer modules work? Anything needed to be added to the Docker build process?

File: panacea/.gitignore (hunk 1 above)
Comment (c-brenn) [reply panacea/.gitignore:28]: they should indeed

**Comment by c-brenn** [comment]
erlang has lex/yacc equivalents built in and mix knows how to automatically compile them.
`mix compile` will create `pml_lexer.erl` and `pml_parser.erl` in the src directory, so the docker build process shouldn't need any changes

File: panacea/test/panacea/pml/parser_test.exs [hunk 2]
```diff
@@ -22,6 +25,17 @@ defmodule Panacea.Pml.ParserTest do
...
     end
 
+    test "it can parse all of jnoll's sample pml" do
+      {:ok, files} = File.ls(@fixtures_dir)
+      for file <-  files do
+        {:ok, pml} = Path.join(@fixtures_dir, file) |> File.read()
+
+        {status, possible_error} = Parser.parse(pml)
+
+        assert status == :ok, "failed to parse #{file}. error: #{inspect possible_error}"
```
Comment (houli) [inline panacea/test/panacea/pml/parser_test.exs:35]: Is this string supposed to mention failure?

Comment (c-brenn) [reply panacea/test/panacea/pml/parser_test.exs:35]: if the assertion fails, the string gets outputted. The normal error message from the generated lexer/parser is very unhelpful - typical erlang. If the parser/lexer reject the pml `possible_error` is `{line_number, lexer/parser module, error_string}`.

Comment (c-brenn) [reply panacea/test/panacea/pml/parser_test.exs:35]: which makes it easier to see what the problem with the lexer/parser is

Comment (houli) [reply panacea/test/panacea/pml/parser_test.exs:35]: Oh excuse me, thought this was some tuple that :ok and an error message and was quite confused

**Pull request merged**